### PR TITLE
Fix machine migrations

### PR DESCRIPTION
**Pull request by SimonRichardson**
The machine domain in the API server was landed too early without thorough testing between 3.6 into main. This prevented me from testing https://github.com/juju/juju/pull/17845 correctly. The solution was to fix two things:

 1. IsController takes a machine Id, not a machine tag.
 2. Sending a raw error to the client. The error must be wrapped in a `ServerError`

In terms of this PR, I modified 2, to just return an empty set of results. This then prevents any required changes to 3.6 branch, and allows us to correctly migrate a 3.6 to 4.0.

We should invest in tests and ensure that we don't break migration 3.6 to 4.0.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
$ juju bootstrap lxd src36
$ juju add-model moveme1
$ juju deploy ubuntu

$ git checkout <this pr>
$ juju bootstrap lxd dst40
$ juju migrate src36:moveme1 dst40
```

## Links

**Jira card:** [JUJU-6411](https://warthogs.atlassian.net/browse/JUJU-6411)

[JUJU-6411]: https://warthogs.atlassian.net/browse/JUJU-6411?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Aflynn50 — APPROVED** [review]
Did QA on this and migration on a model with a few machines in it works fine from 3.6 to 4. 

Title of the PR should match conventional commits.

File: apiserver/facades/client/controller/controller.go [hunk 1]
```diff
@@ -967,7 +967,7 @@ var runMigrationPrechecks = func(
 		}
 	}
 	err = client.Prechecks(ctx, modelInfo)
-	return errors.Annotate(err, "target prechecks failed 2")
+	return errors.Annotate(err, "target prechecks failed")
```
Comment (jack-w-shaw) [inline apiserver/facades/client/controller/controller.go:970]: I'm guessing the change to add "2" to the end of the message here was to distinguish this message from other identical message?

After a grep, I see:
```
$ grep -r "target prechecks failed"
internal/worker/migrationmaster/worker.go:	return errors.Annotate(err, "target prechecks failed")
apiserver/facades/client/controller/controller.go:	return errors.Annotate(err, "target prechecks failed")
```

Perhaps this message should be re-written to be more specific to this specific location, so it isn't confused with the worker?

Comment (SimonRichardson) [reply apiserver/facades/client/controller/controller.go:970]: Yeah, I'll sweep back around to that in another PR.

**Pull request merged**